apps/synapse-portal-framework/src/components/EntityPageMenu.tsx

```diff
@@ -0,0 +1,78 @@
+import { DetailsPageSectionLayoutType } from '@/components/DetailsPage/DetailsPageSectionLayout'
+import { scrollToWithOffset } from '@/utils'
+import { Button, Stack, Typography } from '@mui/material'
+import { NavigateFunction, Location } from 'react-router'
+
+const EntityPageMenu = ({
+  menuSections,
+  navigate,
+  location,
+}: {
+  menuSections: DetailsPageSectionLayoutType[]
+  navigate: NavigateFunction
+  location: Location
```

Comment:
Do these need to be props/any reason you can't just call `useNavigate`/`useLocation` in this component?